backend/src/main/java/com/htecgroup/skynest/service/impl/FolderServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.htecgroup.skynest.exception.folder.FolderNotFoundException;
+import com.htecgroup.skynest.model.dto.FolderDto;
+import com.htecgroup.skynest.model.dto.LoggedUserDto;
+import com.htecgroup.skynest.model.entity.BucketEntity;
+import com.htecgroup.skynest.model.entity.FolderEntity;
+import com.htecgroup.skynest.model.entity.UserEntity;
+import com.htecgroup.skynest.model.request.FolderCreateRequest;
+import com.htecgroup.skynest.model.response.FolderResponse;
+import com.htecgroup.skynest.repository.FolderRepository;
+import com.htecgroup.skynest.repository.UserRepository;
+import com.htecgroup.skynest.service.BucketService;
+import com.htecgroup.skynest.service.CurrentUserService;
+import com.htecgroup.skynest.service.FolderService;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class FolderServiceImpl implements FolderService {
+
+  private ModelMapper modelMapper;
+  private FolderRepository folderRepository;
+  private CurrentUserService currentUserService;
+
+  private UserRepository userRepository;
+
+  private BucketService bucketService;
+
+  @Override
+  public FolderResponse createFolder(FolderCreateRequest folderCreateRequest) {
+    FolderEntity folderEntity = modelMapper.map(folderCreateRequest, FolderEntity.class);
+
+    LoggedUserDto loggedUserDto = currentUserService.getLoggedUser();
+    UserEntity currentUser = userRepository.getById(loggedUserDto.getUuid());
+    BucketEntity bucketEntity =
+        modelMapper.map(
+            bucketService.findBucketByName(folderCreateRequest.getBucketName()),
+            BucketEntity.class);
+
+    if (folderCreateRequest.getParentFolderName() == null
+        || folderCreateRequest.getParentFolderName().isEmpty()) {
+      folderEntity.setParentFolder(null);
+    } else {
+      FolderEntity parentFolderEntity =
+          modelMapper.map(
+              findFolderByName(folderCreateRequest.getParentFolderName()), FolderEntity.class);
+      folderEntity.setParentFolder(parentFolderEntity);
+    }
+
+    folderEntity.setBucket(bucketEntity);
```

Comment:
could users specify a parent folder that's not actually in this bucket?

assuming you use ids instead of names in the request model

Reply:
Well considering I take it from the request, it only depends what get sent as parent folder in the request, so yes.

Reply:
don't forget to check whether the parent folder is in that bucket

or we could just ignore the bucket sent in the request, and just use the parent folder's bucket
if parent folder isn't null

Reply:
I did it with custom annotation